src/scripts/commands/event-reports.ts

```diff
@@ -0,0 +1,59 @@
+import _ from "lodash";
+import { command, flag, subcommands } from "cmd-ts";
+import { getApiUrlOptions, getD2Api } from "scripts/common";
+import { GetCorruptedEventReports } from "domain/usecases/GetCorruptedEventReports";
+import { GetFixedEventReports } from "domain/usecases/GetFixedEventReports";
+import { EventReportsD2Repository } from "data/EventReportsD2Repository";
+import logger from "utils/log";
+import { ExportCorruptedEventReports } from "domain/usecases/ExportCorruptedEventReports";
+import { EventReportsExportFileRepository } from "data/EventReportsExportRepository";
+import { SaveFixedEventReports } from "domain/usecases/SaveFixedEventReports";
+import { UserD2Repository } from "data/UserD2Repository";
+
+export function getCommand() {
+    const corruptedEventReports = command({
+        name: "corrupted event reports",
+        description: "get corrupted event reports",
+        args: {
+            ...getApiUrlOptions(),
+            post: flag({
+                long: "post",
+                description: "Send payload to DHIS2 API. If not present, generates JSON payload.",
+            }),
+        },
+        handler: async args => {
+            const api = getD2Api(args.url);
+            const { post } = args;
+
+            const eventReportsRepository = new EventReportsD2Repository(api);
+            const usersRepository = new UserD2Repository(api);
+            const eventReportsExportRepository = new EventReportsExportFileRepository();
+
+            logger.debug("Fetching corrupted event reports...");
+            const eventReports = await new GetCorruptedEventReports(eventReportsRepository).execute();
+            const fixedEventReports = new GetFixedEventReports(eventReportsRepository).execute(eventReports);
+            logger.info(`Corrupted Event Reports: ${eventReports.length}`);
+
+            const outputFile = "fixed-event-reports.json";
+            new ExportCorruptedEventReports(eventReportsExportRepository).execute(
+                fixedEventReports,
+                outputFile
+            );
+            logger.info(`Written payload: ${outputFile}`);
+
+            if (post) {
```

Comment:
Since the goal of the script is to fix `EventReports` maybe we can have just one usecase `SaveFixedEventReportsUseCase` and then call all the methods of the repositories in there? 

```ts
export class SaveFixedEventReportsUseCase{
// private repos here
    constructor() {}
    
    execute({}): void {
         const eventReports = await this.getCorruptedEventReports();
         const fixedEventReports = await this.getFixedEventReports();
         // all the other methods
    }    
    
    private getCorruptedEventReports() {
        const corruptedEventReports = (await this.eventReportRepository.getAll())
            .filter(ev => !isEmpty(ev.dataElementDimensions))
            .filter(
                ev => !ev.dataElementDimensions.map(pair => pair.programStage.id).includes(ev.programStage.id)
            );

        return corruptedEventReports;
    }
}
```

That way we can have all the logic in one place instead separate use cases. What do you think?

Reply:
I like the suggestion yeah!